portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -57,37 +57,52 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
         int commentsNumber = Integer.parseInt(request.getParameter("number"));
-        
-        List<DataStats> comments = new ArrayList<>();
-        int num = 0;
-        for (Entity entity : results.asIterable()) {
-            long id = entity.getKey().getId();
-            String name = (String) entity.getProperty("name");
-            String comment = (String) entity.getProperty("comment");
-            Date commentTime = (Date) entity.getProperty("commentTime");
-            int upvote = ((Long) entity.getProperty("upvote")).intValue();
-            String imgUrl = (String) entity.getProperty("imgUrl");
-
-            comments.add(new DataStats(name, comment, commentTime, upvote, id, imgUrl));
-            num += 1;
-            if (num == commentsNumber) {
-                break;
+        String json = new String();
+        Gson gson = new Gson(); 
+
+        if (commentsNumber == -1) {
+            // If number is -1, get rating result of all the comments
+            int sum = 0;
+            double totalRating = 0;
+            double[] ratings = new double[7];
```

Comment:
It's unclear what's stored in here. Looking at your code below it seems that indices 0-4 are for the number of ratings 1-5, and the index 5 is for the total number of ratings, and index 6 is for average rating. How about make this ratings array size 5, and then create a numRatings variable and a avgRating variable for what you're currently storing the index 5 and 6? I feel like it's more readable that way.

Reply:
That's true but since I'm returning a json, I'm not sure how to return a list and two variables in this way.

Reply:
There are two solutions as far as I see, one is that you can create a class, which has a ratings array, and two other properties for numRatings and avgRating. Second solution is that you can move the calculation of the average rating to client side, and only return the list of 5 ratings here. Personally I would go with the second solution, since I feel like calculating the average is quite straightforward.

Reply:
Sounds good!